scripts/skills/perks/perk_rf_second_wind.nut

```diff
@@ -0,0 +1,31 @@
+this.perk_rf_second_wind <- ::inherit("scripts/skills/skill", {
+	m = {
+		RecoverActionPointTarget = 4
+	},
+	function create()
+	{
+		this.m.ID = "perk.rf_second_wind";
+		this.m.Name = ::Const.Strings.PerkName.RF_SecondWind;
+		this.m.Description = ::Const.Strings.PerkDescription.RF_SecondWind;
+		// this.m.Icon = "ui/perks/rf_second_wind.png";		// TODO: add icon for this perk
+		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Order = ::Const.SkillOrder.Perk - 5;
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsHidden = false;
+	}
+
+	function onWaitTurn()
+	{
+		local actor = this.getContainer().getActor();
+
+		local recoveredActionPoints = this.m.RecoverActionPointTarget - actor.getActionPoints();
+		if (recoveredActionPoints > 0)
+		{
+			actor.setActionPoints(actor.getActionPoints() + recoveredActionPoints);
```

Comment:
You need to make sure you don't exceed the maximum action points (in case for some weird reason this actor's max action points are fewer than 4).